app/main.py

```diff
@@ -1,2 +1,128 @@
+from __future__ import annotations
+from typing import Any, Iterable, Hashable
+
+
+_not_provided = object()
+
+
 class Dictionary:
-    pass
+    DEFAULT_SIZE = 8
+    THRESHOLD = 2 / 3
+
+    def __init__(self, items: Iterable = (), **kwargs) -> None:
+        self.length = 0
+        self.hash_table = [None] * self.DEFAULT_SIZE
+
+        for key, value in items:
+            self[key] = value
+        self.update(kwargs)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        _, _, value = self.hash_table[self._get_with_hash(key, hash(key))]
```

Comment:
We actually looking for a different solution for hash table - you can create a Node class to store necessary data - it will structure your code better

   

>  Also, don't forget to store (key, hash, value) as a node in the hash table. A hash table may be stored as a simple list of nodes.